etc/kayobe/ansible/get-cloud-facts-v2.yml

```diff
@@ -0,0 +1,114 @@
+---
+- name: Can the control host reach the internet?
+  hosts: localhost
+  gather_facts: true
+  tasks:
+    - name: Check internet connectivity
+      ansible.builtin.shell:
+        cmd: "ping stackhpc.com -c 3"
+      register: internet_connectivity
+      timeout: 5
+      ignore_errors: true
+    - name: Set internet connectivity fact
+      set_fact:
+        internet_connectivity: "{{ not internet_connectivity.failed }}"
+
+- name: Get facts for seed node
+  hosts: seed
+  gather_facts: true
+  tasks:
+
+    - name: Is there a dedicated seed node?
+      set_fact:
+        seed_node_is_vm: "{{ ansible_facts.virtualization_role == 'guest' }}"
+
+- name: Get facts
+  hosts: all
+  gather_facts: true
+  tasks:
+    - name: Get OS distribution
+      set_fact:
+        distribution: "{{ ansible_facts.distribution }}"
+    - name: Get OS distribution release
+      set_fact:
+        distribution_release: "{{ ansible_facts.distribution_release }}"
+    - name: Get kernel version
+      ansible.builtin.shell:
+        cmd: "uname -r"
+      register: kernel_version
+    - name: Get running contianers
+      ansible.builtin.shell:
+        cmd: "docker ps --format {% raw %}'{{.Names}}'{% endraw %}"
+      become: true
+      register: containers_list
+- name: Get facts for computes
+  hosts: compute
+  gather_facts: true
+  tasks:
+    - name: Check if hugepages enabled
+      set_fact:
+        hugepages_enabled: "{{ ansible_facts.cmdline }}"
+
+- name: Gather Cloud Facts
+  hosts: localhost
+  gather_facts: true
+  tasks:
+    - name: Write facts to file
+      vars:
+        cloud_facts:
+          internet_connectivity: "{{ internet_connectivity }}"
+          seed_node_is_vm: "{{ hostvars[groups['seed'][0]]['seed_node_is_vm'] }}"
+          distributions: "{{ hostvars | group_hostvars_by_var('distribution') }}"
+          dist_releases: "{{ hostvars | group_hostvars_by_var('distribution_release') }}"
+          kernel_versions: "{{ hostvars | group_hostvars_by_var('kernel_version','stdout') }}"
+          deployed_containers: "{{ hostvars | group_hostvars_by_host('containers_list', 'stdout') }}"
```

Comment:
group_hostvars_by_host is the new filter right? You appear to be missing it in this PR